diode-server/netbox/virtualization_wrappers.go

```diff
@@ -0,0 +1,1680 @@
+		if err != nil {
+			return nil, err
+		}
+		virtualMachine.Tags = nil
+
+		if !actualVirtualMachine.HasChanged() {
+			virtualMachine = &VirtualizationVirtualMachine{
+				ID: actualVirtualMachine.ID(),
+			}
+
+			intendedVirtualMachineID := intendedVirtualMachine.ID()
+			if intended.VirtualInterface.VirtualMachine != nil {
+				intendedVirtualMachineID = intended.VirtualInterface.VirtualMachine.ID
+			}
+
+			intended.VirtualInterface.VirtualMachine = &VirtualizationVirtualMachine{
+				ID: intendedVirtualMachineID,
+			}
+		}
+
+		vw.VirtualInterface.VirtualMachine = virtualMachine
+
+		vw.objectsToReconcile = append(vw.objectsToReconcile, virtualMachineObjectsToReconcile...)
+
+		if vw.VirtualInterface.Enabled == nil {
+			vw.VirtualInterface.Enabled = intended.VirtualInterface.Enabled
+		}
+
+		if vw.VirtualInterface.MTU == nil {
+			vw.VirtualInterface.MTU = intended.VirtualInterface.MTU
+		}
+
+		if vw.VirtualInterface.MACAddress == nil {
+			vw.VirtualInterface.MACAddress = intended.VirtualInterface.MACAddress
+		}
+
+		if vw.VirtualInterface.Description == nil {
+			vw.VirtualInterface.Description = intended.VirtualInterface.Description
+		}
+
+		tagsToMerge := mergeTags(vw.VirtualInterface.Tags, intended.VirtualInterface.Tags, intendedNestedObjects)
+
+		if len(tagsToMerge) > 0 {
+			vw.VirtualInterface.Tags = tagsToMerge
+		}
+
+		for _, t := range vw.VirtualInterface.Tags {
+			if t.ID == 0 {
+				vw.objectsToReconcile = append(vw.objectsToReconcile, &TagDataWrapper{Tag: t, hasParent: true})
+			}
+		}
+
+		actualHash, _ := hashstructure.Hash(vw.Data(), hashstructure.FormatV2, nil)
+		intendedHash, _ := hashstructure.Hash(intended.Data(), hashstructure.FormatV2, nil)
+
+		reconciliationRequired = actualHash != intendedHash
+	} else {
+		virtualMachineObjectsToReconcile, virtualMachineErr := actualVirtualMachine.Patch(intendedVirtualMachine, intendedNestedObjects)
+		if virtualMachineErr != nil {
+			return nil, virtualMachineErr
+		}
+
+		virtualMachine, err := copyData(actualVirtualMachine.Data().(*VirtualizationVirtualMachine))
+		if err != nil {
+			return nil, err
+		}
+		virtualMachine.Tags = nil
+
+		if !actualVirtualMachine.HasChanged() {
+			virtualMachine = &VirtualizationVirtualMachine{
+				ID: actualVirtualMachine.ID(),
+			}
+		}
+		vw.VirtualInterface.VirtualMachine = virtualMachine
+
+		vw.objectsToReconcile = append(vw.objectsToReconcile, virtualMachineObjectsToReconcile...)
+
+		tagsToMerge := mergeTags(vw.VirtualInterface.Tags, nil, intendedNestedObjects)
+
+		if len(tagsToMerge) > 0 {
+			vw.VirtualInterface.Tags = tagsToMerge
+		}
+
+		for _, t := range vw.VirtualInterface.Tags {
+			if t.ID == 0 {
+				vw.objectsToReconcile = append(vw.objectsToReconcile, &TagDataWrapper{Tag: t, hasParent: true})
+			}
+		}
+	}
+
+	if reconciliationRequired {
+		vw.hasChanged = true
+		vw.objectsToReconcile = append(vw.objectsToReconcile, vw)
+	}
+
```

Comment:
To ensure we don't have duplicated objects to reconcile, we should use `dedupObjectsToReconcile` func:
```suggestion

        dedupObjectsToReconcile, err := dedupObjectsToReconcile(vw.objectsToReconcile)
	if err != nil {
		return nil, err
	}
	vw.objectsToReconcile = dedupObjectsToReconcile

```